docs/documentation/securing_apps/topics/oidc/supported-grant-types.adoc

```diff
@@ -0,0 +1,105 @@
+Token. This results in the need to either permit Access Tokens with a long expiration, which is problematic as it's very hard to invalidate these. Or
+requires a new redirect to obtain new Access Token once the initial Access Token has expired. The Implicit flow is useful if the application only wants to
+authenticate the user and deals with logout itself.
+
+There's also a Hybrid flow where both the Access Token and an Authorization Code is returned.
+
+One thing to note is that both the Implicit flow and Hybrid flow has potential security risks as the Access Token may be leaked through web server logs and
+browser history. This is somewhat mitigated by using short expiration for Access Tokens.
+
+For more details refer to the https://openid.net/specs/openid-connect-core-1_0.html#ImplicitFlowAuth[Implicit Flow] in the OpenID Connect specification.
+
+[[_resource_owner_password_credentials_flow]]
+==== Resource Owner Password Credentials
+
+Resource Owner Password Credentials, referred to as Direct Grant in {project_name}, allows exchanging user credentials for tokens. It's not recommended
+to use this flow unless you absolutely need to. Examples where this could be useful are legacy applications and command-line interfaces.
+
+There are a number of limitations of using this flow, including:
+
+* User credentials are exposed to the application
+* Applications need login pages
+* Application needs to be aware of the authentication scheme
+* Changes to authentication flow requires changes to application
+* No support for identity brokering or social login
+* Flows are not supported (user self-registration, required actions, etc.)
+
+For a client to be permitted to use the Resource Owner Password Credentials grant the client has to have the `Direct Access Grants Enabled` option enabled.
+
+This flow is not included in OpenID Connect, but is a part of the OAuth 2.0 specification.
+
+For more details refer to the https://datatracker.ietf.org/doc/html/rfc6749#section-4.3[Resource Owner Password Credentials Grant] chapter in the OAuth 2.0 specification.
+
+===== Example using CURL
+
+The following example shows how to obtain an access token for a user in the realm `master` with username `user` and password `password`. The example is using
+the confidential client `myclient`:
+
+[source,bash,subs="attributes+"]
+----
+curl \
+  -d "client_id=myclient" \
+  -d "client_secret=40cc097b-2a57-4c17-b36a-8fdf3fc2d578" \
+  -d "username=user" \
+  -d "password=password" \
+  -d "grant_type=password" \
+  "http://localhost:8080{kc_realms_path}/master/protocol/openid-connect/token"
+----
+
+==== Client credentials
+
+Client Credentials is used when clients (applications and services) wants to obtain access on behalf of themselves rather than on behalf of a user. This can
+for example be useful for background services that applies changes to the system in general rather than for a specific user.
```

Comment:
```suggestion
Client Credentials are used when clients (applications and services) want to obtain access on behalf of themselves rather than on behalf of a user. These credentials can
for example be useful for background services that apply changes to the system in general rather than for a specific user.
```